internal/metrics/metrics.go

```diff
@@ -20,42 +31,58 @@ var (
 		Name:      discoveryInstanceDurations,
 		Help:      "Instance discovery latencies in seconds",
 		Buckets:   discoveryInstanceDurationsBuckets,
-	}, []string{"cluster_name", "hostname"})
+	}, []string{labelClusterName, labelHostName})
 
 	discoveryClusterDurationsSum = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Subsystem: "discovery",
 		Name:      discoveryClusterDurations,
 		Help:      "Cluster discovery latencies in seconds",
 		Buckets:   discoveryClusterDurationsBuckets,
-	}, []string{"cluster_name"})
+	}, []string{labelClusterName})
 
 	shardCriticalLevelGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Subsystem: "shard",
 		Name:      shardCriticalLevel,
 		Help:      "Critical level of the replica set",
-	}, []string{"cluster_name", "uuid", "master_uri"})
+	}, []string{labelClusterName, labelShardUUID})
 
 	shardStateGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Subsystem: "shard",
 		Name:      shardState,
 		Help:      "The state of each shard in the cluster; it will have one line for each possible state of each shard. A value of 1 means the shard is in the state specified by the state label, a value of 0 means it is not.",
-	}, []string{"cluster_name", "uuid", "master_uri", "state"})
+	}, []string{labelClusterName, labelShardUUID, labelShardState})
 
 	discoveryErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Subsystem: "discovery",
 		Name:      "errors",
 		Help:      "Errors that happen during discovery process",
-	}, []string{"cluster_name", "uri"})
+	}, []string{labelClusterName, labelURI})
+
+	recoveryEvents = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem: "orchestrator",
+		Name:      recoveryEvent,
+		Help:      "Registered shard recovery events",
+	}, []string{labelClusterName, labelShardUUID, labelShardState})
 )
 
 func init() {
-	discoveryErrors.With(prometheus.Labels{"cluster_name": "", "uri": ""}).Add(0)
+	recoveryEvents.With(prometheus.Labels{
```

Comment:
You'll have metrics with empty labels, don't think it is a good idea.

You should initialize metrics with actual cluster, shard, instance labels like it is done for shard state metric.